catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -19,9 +21,29 @@ import useMemoEq from 'utils/useMemoEq'
 import * as Types from 'utils/types'
 
 import EditFileMeta from './EditFileMeta'
+import EditFileName from './EditFileName'
 import * as PD from './PackageDialog'
 import * as S3FilePicker from './S3FilePicker'
 
+function resolveNameConflictRudely(name: string, attempt: number) {
+  const ext = extname(name)
+  const base = name.split(ext)[0]
+  return `${base} (conflict ${attempt})${ext}`
+}
+
+function resolveName(
+  name: string,
+  obj: Record<string, any>,
+  attempt: number = 1,
+): string {
+  if (!obj[name]) return name
+
+  const newName = resolveNameConflictRudely(name, attempt)
+  if (!obj[newName]) return newName
+
+  return resolveName(name, obj, attempt + 1)
```

Comment:
i'm not sure what sacrifice do you mean:
```suggestion
function resolveName(name: string, obj: Record<string, any>): string {
  let newName = name, attempt = 1
  while (newName in obj) newName = resolveNameConflictRudely(name, attempt++)
  return newName
```